frontend/src/console.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from "react";
+import { BrowserRouter } from "react-router-dom"
+import Page from "./components/Page";
+import userContext from "./contexts/context";
+
+const jwt_decode = require("jwt-decode");
+
+function Console(props) {
+    const [user, setUser] = useState({});
+    const [auth, setAuth] = useState(false);
+    const [admin, setAdmin] = useState(false);
+    const [currentUrl, setUrlState] = useState(() => '/');
+
+    function setUserAndAuth() {
+      const decodedUser = jwt_decode(localStorage.jwtToken);
+      setUser(decodedUser);
+      setAuth(true);
+      setAdmin(decodedUser.isAdmin);
+      const currentTime = Date.now() / 1000;
+      if (decodedUser.exp < currentTime) {
+        localStorage.removeItem("jwtToken");
+        setUser({});
+        setAuth(false);
+        setAdmin(false);
```

Comment:
you have a flow here that sets auth & admin to ture and then to false. This is not good. always have one set - calculate the value you want to set before. and then call `setAdmin` once, `setAuth` once, etc